CONTRIBUTING.rst

```diff
@@ -63,6 +63,32 @@ a PR until it conforms to the requirements described here (e.g.
 coding style, release notes, etc.) and it is confirmed that the code
 has sufficient unit tests and does not break any existing unit tests.
 
+Commits
+-------
+
+While not a strict requirement, it is beneficial for commits to record a set of
+separate, logical changes.  Ideally, each commit should compile and have
```

Comment:
Ideally may be too soft but I can see why you said that on a per commit basis.

Reply:
In my experience, the ideal of small, easy to understand commits can sometimes end up at odds with a repo that is functional at every commit. Usually, this will happen during a major refactor or addition; it's not *that* uncommon for "refactor the thing" and "fix all the users of the thing" to end up as separate commits, especially if some of the uses are code that isn't enabled by default. Seeing as I've made such commits, I was trying to avoid leaning too heavily into "do as I say, not as I do" territory :slightly_smiling_face:.

Reply:
I agree that "should"  is better than "shall" here. I am usually in compliance, but there are cases where I have organized the commits where some won't build.